### PR TITLE
Create a snapshot of the workspace for users to load upon connecting.

### PR DESCRIPTION
**Pull request by navilperez**
Create and store a snapshot of the current state of the workspace
whenever a new user connects. A snapshot object contains the XML of the
workspace and the serverId of the last entry represented on the
workspace.

### CODE REVIEW

File: blockly-rtc/src/WorkspaceClient.js [hunk 1]
```diff
@@ -48,15 +49,22 @@ export default class WorkspaceClient {
...
    * @public
    */
-  async initiateWorkspace() {
-    // TODO: Get an SVG of the current blocks on the workspace instead of
-    // replaying history.
-    const events = await this.getEventsHandler(0);
+  async start() {
+    // Load the current state of the workspace.
+    const workspace = Blockly.Workspace.getById(this.workspaceId);
+    const snapshot = await this.getSnapshotHandler();
```
Comment (rachel-fenichel) [inline blockly-rtc/src/WorkspaceClient.js:59]: Will snapshot ever be null?

**Review by rachel-fenichel — COMMENTED** [review]
One Q, mostly lgtm

File: blockly-rtc/src/WorkspaceClient.js (hunk 1 above)
Comment (navilperez) [reply blockly-rtc/src/WorkspaceClient.js:59]: It should not be, the server has a valid default value if it has not been set. I can only think of the case where there are errors retrieving it from the server in which case this.getSnapshotHandler() will reject. Is this a case I should account for?

Comment (rachel-fenichel) [reply blockly-rtc/src/WorkspaceClient.js:59]: Nope, that sounds fine.

**Review by rachel-fenichel — APPROVED** [review]

**Pull request merged**